.github/workflows/required.yml

```diff
@@ -3,24 +3,40 @@ name: Required Tests
 on: [push, pull_request]
 
 jobs:
-  test:
+  build-test:
+    name: Build Test
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        image:
-          - 'fedora_latest_jdk11'
-          - 'debian_jdk11'
-          - 'ubuntu_jdk11'
-          # Disable tests due to missing dependencies
-          # - 'centos_7'
-          # - 'centos_8'
-
+        os:
+          - 'fedora:latest'
+          - 'debian:testing'
+          - 'ubuntu:rolling'
+          # Disable CentOS due to missing dependencies
+          # - 'centos:7'
+          # - 'centos:8'
+    container: ${{ matrix.os }}
     steps:
-    - name: Clone the repository
+    - name: Clone repository
       uses: actions/checkout@v2
 
-    - name: Build and Run the Docker Image
-      run: bash tools/run_container.sh "${{ matrix.image }}"
+    - name: Install Fedora/CentOS dependencies
+      if: ${{ startsWith(matrix.os, 'fedora:') || startsWith(matrix.os, 'centos:') }}
+      run: |
+        dnf install -y dnf-plugins-core rpm-build
+        dnf builddep -y --spec jss.spec
+
+    - name: Install Debian/Ubuntu dependencies
+      if: ${{ startsWith(matrix.os, 'debian:') || startsWith(matrix.os, 'ubuntu:') }}
+      run: |
+        apt-get update
+        apt-get install -y \
```

Comment:
This is fewer dependencies than specified in either of the old Docker images. I'm happy with it, we obviously don't need them and removing dead code is good, I'm just curious how you knew to remove them.

Reply:
@ckelleyRH Thanks! I didn't actually expect to remove anything, but when there's a static list like this sometimes we just need to revalidate it. So I started from scratch and readd the dependencies if they are still needed.